src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +133,51 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        Object instance = null;
+        // if the name is a simple class name, try to find it in maps at first
+        if (!name.contains(".")) {
+            instance = createObjectFromMap(name, NAME_TO_FULL_MODULE_NAME);
+            if (instance == null) {
+                if (thirdPartyNameToFullModuleName == null) {
+                    thirdPartyNameToFullModuleName =
+                            generateThirdPartyNameToFullModuleName(moduleClassLoader);
```

Comment:
@romani I want to share the results of `generateThirdPartyNameToFullModuleName` through the field `thirdPartyNameToFullModuleName`. This is all modules we recognize in the classpath.
We are doing similar scanning at [sevntu](https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/internal/CheckUtil.java#L116) and [sonar](https://github.com/checkstyle/sonar-checkstyle/blob/master/checkstyle-sonar-plugin/src/test/java/org/sonar/plugins/checkstyle/internal/CheckUtil.java#L61). We could swap out the scans there and use `PackageObjectFactory` directly.
This would remove reliance on guava in other projects, but we would still use guava in CS for this scan.